From e0b4f943bbe7fc82f0eaad13042816f3d70a8493 Mon Sep 17 00:00:00 2001
From: cnauroth <cnauroth@apache.org>
Date: Wed, 1 Jul 2015 19:47:58 -0700
Subject: [PATCH 0668/1023] HADOOP-12172. FsShell mkdir -p makes an
 unnecessary check for the existence of the
 parent. Contributed by Chris Nauroth.

(cherry picked from commit f3796224bfdfd88e2428cc8a9915bdfdc62b48f3)
(cherry picked from commit 0d94a293a8baa4f4eaeb7015b6e23de2286a20e2)
---
 .../java/org/apache/hadoop/fs/shell/Mkdir.java     |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Mkdir.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Mkdir.java
index 74bad62..9f39da2 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Mkdir.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/Mkdir.java
@@ -70,7 +70,8 @@ protected void processPath(PathData item) throws IOException {
   protected void processNonexistentPath(PathData item) throws IOException {
     // check if parent exists. this is complicated because getParent(a/b/c/) returns a/b/c, but
     // we want a/b
-    if (!item.fs.exists(new Path(item.path.toString()).getParent()) && !createParents) {
+    if (!createParents &&
+        !item.fs.exists(new Path(item.path.toString()).getParent())) {
       throw new PathNotFoundException(item.toString());
     }
     if (!item.fs.mkdirs(item.path)) {
-- 
1.7.9.5

